crypto.go

```diff
@@ -324,56 +372,64 @@ type x25519PublicKey struct {
 }
 
 type x25519Scheme struct {
+	KDF KDFScheme
 	skE KEMPrivateKey
 }
 
+func (s x25519Scheme) internalKDF() KDFScheme {
+	return s.KDF
+}
+
 func (s x25519Scheme) ID() KEMID {
 	return DHKEM_X25519
 }
 
-func (s x25519Scheme) GenerateKeyPair(rand io.Reader) (KEMPrivateKey, KEMPublicKey, error) {
-	priv := &x25519PrivateKey{}
-	_, err := rand.Read(priv.val[:])
+func (s x25519Scheme) DeriveKeyPair(ikm []byte) (KEMPrivateKey, KEMPublicKey, error) {
+	idBuffer := make([]byte, 2)
+	binary.BigEndian.PutUint16(idBuffer, uint16(s.ID()))
+	dkp_prk := s.KDF.LabeledExtract(nil, string(idBuffer)+"dkp_prk", ikm)
```

Comment:
Another question about DeriveKeyPair: is `I2OSP(idBuffer) + "dpk_prk"` the label you want to use? The spec says `I2OSP(kem_id, 2)`.

Reply:
I'm going off the changes in this to-be-landed PR: https://github.com/cfrg/draft-irtf-cfrg-hpke/pull/122

Reply:
Aha, thanks for the link.